From 802d65c06a02f6c52aa4996f33cf414afa99c33b Mon Sep 17 00:00:00 2001
From: Artem Senichev <a.senichev@yadro.com>
Date: Wed, 5 Jun 2019 14:33:47 +0300
Subject: [PATCH] Increase buffer size for writing FIRDATA

Temporary workaround, see
https://github.com/open-power/occ/issues/24

Unlike the old method of writing FIRDATA, using HIOMAP takes more
time to do the same.
OCC doesn't have enough time to perform all operations, something
stops all activity on the CPU before OCC completes its work. As a
result, we have only the first part of FIRDATA.

Increasing the PAGE_PROGRAM_BYTES parameter reduces IPMI traffic
and FIRDATA will be filled without ECC errors.

Signed-off-by: Artem Senichev <a.senichev@yadro.com>
---
 src/occ/firdata/norflash.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/occ/firdata/norflash.h b/src/occ/firdata/norflash.h
index d41c878..5da050d 100644
--- a/src/occ/firdata/norflash.h
+++ b/src/occ/firdata/norflash.h
@@ -95,7 +95,7 @@ typedef enum
  */
 enum
 {
-    PAGE_PROGRAM_BYTES = 256, /***< 256 bytes per PP command */
+    PAGE_PROGRAM_BYTES = 1024, /***< 1024 bytes per PP command */
 };
 
 
-- 
2.21.0

